src/Chapter3.hs

```diff
@@ -971,6 +1040,20 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data Weekday = Mon | Tue | Wed | Thu | Fri | Sat | Sun deriving (Enum)
+
+isWeekend :: Weekday -> Bool
+isWeekend Sat = True
+isWeekend Sun = True
+isWeekend _ = False
+
+nextDay :: Weekday -> Weekday
+nextDay = succ
+
+daysToParty :: Weekday -> Int
+daysToParty Fri = 0
+daysToParty day = (1 +) $ daysToParty $ nextDay day
```

Comment:
Nice recursive implementation! Though, it's possible to provide a more efficient one by using the `fromEnum` method of the `Enum` typeclass